### PR TITLE
Use nix based docker base image

### PR DESCRIPTION
**Pull request by sveitser**
### Motivation

1. No need for patchelf. Base image inputs are tracked with nix.
2. Make it possible to build the image locally. Should make it easier to put together the remaining images we need.

### CODE REVIEW

**Comment by sveitser** [comment]
Building the base image works on the CI bu the final "Build and Push Docker" step fails. I think it's because it's trying to fetch metadata about the base image from dockerhub and that image doesn't exist.

https://github.com/EspressoSystems/cape/runs/5511464159?check_suite_focus=true

**Comment by sveitser** [comment]
Possibly relevant: https://github.com/docker/build-push-action/issues/425#issuecomment-892807549

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -118,6 +117,9 @@ jobs:
         with:
           images: ghcr.io/espressosystems/cape/wallet
 
+      - name: Generate base image
+        run: nix-shell --run build-docker-base
```
Comment (sveitser) [inline .github/workflows/build.yml:121]: This is a bit wasteful and we don't really need to activate the nix shell but I'm not the `nix build ...` command works without it.

Comment (sveitser) [reply .github/workflows/build.yml:121]: Will give it a go when I get around to it.

**Review by Ancient123 — APPROVED** [review]

**Review by DieracDelta — APPROVED** [review]

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -118,6 +117,9 @@ jobs:
         with:
           images: ghcr.io/espressosystems/cape/wallet
 
+      - name: Generate base image
+        run: bash bin/build-docker-base
```
Comment (Ancient123) [inline .github/workflows/build.yml:121]: This is gonna actually be `wallet-checkout/bin/build-docker-base` for weird permissions reasons.

**Review by Ancient123 — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 2 above)
Comment (sveitser) [reply .github/workflows/build.yml]: @Ancient123 seems to work now.

File: .github/workflows/build.yml (hunk 1 above)
Comment (sveitser) [reply .github/workflows/build.yml]: This is done. The docker workflow just takes seconds now.

**Review by Ancient123 — APPROVED** [review]

**Pull request merged**